tasks/orca_catalog_reporting/orca_catalog_reporting.py

```diff
@@ -118,19 +118,19 @@ def get_catalog_sql() -> text:
                 granules.id,
                 granules.collection_id, 
                 granules.cumulus_granule_id, 
-                EXTRACT(EPOCH FROM granules.cumulus_create_time AT TIME ZONE 'UTC') * 1000 as cumulus_create_time, 
+                (EXTRACT(EPOCH FROM date_trunc('milliseconds', granules.cumulus_create_time) AT TIME ZONE 'UTC') * 1000)::bigint as cumulus_create_time, 
                 granules.execution_id, 
-                EXTRACT(EPOCH FROM granules.ingest_time AT TIME ZONE 'UTC') * 1000 as ingest_time, 
-                EXTRACT(EPOCH FROM granules.last_update AT TIME ZONE 'UTC') * 1000 as last_update
+                (EXTRACT(EPOCH FROM date_trunc('milliseconds', granules.ingest_time) AT TIME ZONE 'UTC') * 1000)::bigint as ingest_time, 
+                (EXTRACT(EPOCH FROM date_trunc('milliseconds', granules.last_update) AT TIME ZONE 'UTC') * 1000)::bigint as last_update
             FROM
             (SELECT DISTINCT
                 granules.cumulus_granule_id
             FROM granules
             WHERE 
                 (:granule_id is null or cumulus_granule_id=ANY(:granule_id)) and 
                 (:collection_id is null or collection_id=ANY(:collection_id)) and 
-                (:start_timestamp is null or EXTRACT(EPOCH FROM cumulus_create_time AT TIME ZONE 'UTC') * 1000>=:start_timestamp) and 
-                (:end_timestamp is null or EXTRACT(EPOCH FROM cumulus_create_time AT TIME ZONE 'UTC') * 1000<:end_timestamp)
+                (:start_timestamp is null or EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC') * 1000>=:start_timestamp) and 
```

Comment:
A little nit picky but might not hurt to cast the extract here to bigint like done in other SQL to make the datatype consistent. Driver will still handle the comparison the same.

Reply:
Changed